src/jdk.jfr/share/classes/jdk/jfr/internal/instrument/FileChannelImplInstrumentor.java

```diff
@@ -90,22 +100,30 @@ public int read(ByteBuffer dst) throws IOException {
     @JIInstrumentationMethod
     public int read(ByteBuffer dst, long position) throws IOException {
         FileReadEvent event = FileReadEvent.EVENT.get();
-        if (!event.isEnabled()) {
-            return read(dst, position);
-        }
+        FileReadIOStatisticsEvent readPeriodicEvent = FileReadIOStatisticsEvent.EVENT.get();
         int bytesRead = 0;
-        try {
-            event.begin();
+
+        long startTime = System.nanoTime();
+        if (!event.isEnabled()) {
             bytesRead = read(dst, position);
-        } finally {
-            if (bytesRead < 0) {
-                event.endOfFile = true;
-            } else {
-                event.bytesRead = bytesRead;
+        } else if (event.isEnabled()) {
+            try {
+                event.begin();
+                bytesRead = read(dst, position);
+            } finally {
+                if (bytesRead < 0) {
+                    event.endOfFile = true;
+                } else {
+                    event.bytesRead = bytesRead;
+                }
+                event.path = path;
+                event.commit();
+                event.reset();
             }
-            event.path = path;
-            event.commit();
-            event.reset();
+        }
+        if (readPeriodicEvent.isEnabled()) {
+            long duration = System.nanoTime() - startTime;           
```

Comment:
I think this Is good, ie we remove extra calls to nanoTime() If the periodic event is not enabled - but we will lose some timing information as as of now, the time potentially includes the time to commit the FileRead/Write events